eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/EventsourcedProcessor.scala

```diff
@@ -113,7 +120,9 @@ trait EventsourcedProcessor extends EventsourcedWriter[Long, Long] with ActorLog
     case payload if processEvent.isDefinedAt(payload) =>
       val currentProcessedEvents = processEvent(payload)
       if (lastSequenceNr > processingProgress)
-        processedEvents = processedEvents :+ currentProcessedEvents.map(createEvent(_, lastHandledEvent.customDestinationAggregateIds))
+        processedEvents = processedEvents :+ currentProcessedEvents.map { e =>
+          postProcessDurableEvent(createEvent(e, lastHandledEvent.customDestinationAggregateIds))
```

Comment:
Taking a look at the fields in DurableEvent I'd say there are maybe more than these 3 fields you mentioned that shouldn't be changed:

```scala
case class DurableEvent(
  payload: Any,
  emitterId: String = DurableEvent.UndefinedEmittedId,
  emitterAggregateId: Option[String] = None,
  customDestinationAggregateIds: Set[String] = Set(),
  systemTimestamp: Long = 0L,
  vectorTimestamp: VectorTime = VectorTime.Zero,
  processId: String = DurableEvent.UndefinedLogId,
  localLogId: String = DurableEvent.UndefinedLogId,
  localSequenceNr: Long = DurableEvent.UndefinedSequenceNr,
  deliveryId: Option[String] = None,
  persistOnEventSequenceNr: Option[Long] = None)
```

Maybe I'm missing some use-cases here but I'm not exactly sure which other fields than `customDestinationAggregateIds` should be allowed to change in the first place:

- `payload` is "created" in `processEvent`, no need to allow to post-process it
- `emitterId` could be allowed to change, even though I don't have a use-case in mind now
- `systemTimestamp` should be set by the `EventLog` implementation as I now learned in the context of #370, right? If so, it shouldn't be allowed to be changed here, either.
- `vectorTimestamp` should not change as you said
- `processId` should not change as you said
- `localLogId` probably not !?
- `localSequenceNr` probably not !?
- `deliveryId` probably not !?
- `persistOnEventSequenceNr` probably not !?

The assumption of my initial implementation was "garbage in, garbage out", i.e. the implementor of the processor should know what he/she is doing. Therefore, I also didn't add a test because would be no real assumptions that can be verified (other than e.g. the use-case of changing the customDestinationAggregateId which I could obviously add).

WDYT? If nothing should be allowed to change other than `customDestinationAggregateIds` (and `emitterAggregateId`?) maybe it would be sensible to add something specific for that feature.